File: 20220727/모범답안/20220727_10.js/CombineTwoDigitOddNumber_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.hard.CombineTwoDigitEvenNumber.CombineTwoDigitEvenNumber import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        array = [1, 3, 4]

        # when
        result = solution(array)

        # then
        self.assertEqual(4, result)

    def test_2(self):
        # given
        array = [1, 4, 6, 8]

        # when
        result = solution(array)

        # then
        self.assertEqual(3, result)

    def test_3(self):
        # given
        array = [4, 2, 6, 8, 4, 2]

        # when
        result = solution(array)

        # then
        self.assertEqual(0, result)

    def test_4(self):
        # given
        array = []

        # when
        result = solution(array)

        # then
        self.assertEqual(0, result)

    def test_5(self):
        # given
        array = [5]

        # when
        result = solution(array)

        # then
        self.assertEqual(0, result)

    def test_6(self):
        # given
        array = [5, 5]

        # when
        result = solution(array)

        # then
        self.assertEqual(1, result)


if __name__ == '__main__':
    unittest.main()

File: 20220727/모범답안/20220727_10.js/CombineTwoDigitOddNumber_test.js
import assert from 'assert'
import solution from './CombineTwoDigitOddNumber.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const array = [1, 3, 4]

    // when
    const result = solution(array)

    // then
    assert.deepStrictEqual(4, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const array = [1, 4, 6, 8]

    // when
    const result = solution(array)

    // then
    assert.deepStrictEqual(3, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const array = [4, 2, 6, 8, 4, 2]

    // when
    const result = solution(array)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const array = []

    // when
    const result = solution(array)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const array = [5]

    // when
    const result = solution(array)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const array = [5, 5]

    // when
    const result = solution(array)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });
})
